packages/waas/src/intents/transactions.ts

```diff
@@ -104,6 +187,31 @@ export function getTransactionReceipt({
   })
 }
 
+export function sendERC20ArgsToTransaction({ token, to, value, ...args }: Omit<SendERC20Args, 'chainId'>): Transaction {
+  return erc20({ tokenAddress: token, to, value: value.toString() })
+}
+
+export function sendERC721ArgsToTransaction({ token, to, id, safe, data, ...args }: Omit<SendERC721Args, 'chainId'>): Transaction {
+  return erc721({ tokenAddress: token, to, id, data, safe })
+}
+
+export function sendERC1155ArgsToTransaction({ token, to, values, data, ...args }: Omit<SendERC1155Args, 'chainId'>): Transaction {
+  const vals = values.map(v => ({
+    id: v.id,
+    amount: ethers.BigNumber.from(v.amount).toString()
+  }))
+
+  return erc1155({ tokenAddress: token, to, vals, data })
+}
+
+export function sendDelayedEncodeArgsToTransaction({ to, value, abi, func, args, ...otherArgs }: Omit<SendDelayedEncodeArgs, 'chainId'>): Transaction {
+  return delayedEncode({
+    to,
+    value: ethers.BigNumber.from(value).toString(),
+    data: { abi, func, args }
+  })
+}
```

Comment:
These functions look quite unwieldy - the base `erc20`, `erc721`, etc. functions were supposed to be the ones used by library consumers to easily turn those into `Transaction`. Now we have another way to do it which is probably wrong...

Can we instead move this functionality to the simple `erc20` etc. functions? E.g. so that they support both of the formats and consumers aren't confused which to use.

Reply:
Unfortunately, those functions were never public, their params are different(not only names but also types). In addition to that function names such as ``erc20``, ``erc1155``... do not really say much about what these functions do.

My first approach just broke SDK API contract but was clean. This one preserves API but is not as clean. We could alter ``erc20``... functions so they work with bot params but again it will not be very clean and intuitive to use.

@taylanpince @pkieltyka WDYT?

Reply:
I prefer the cleaner method here. Nobody is using this part of the SDK on production yet, this is still a good time to make a breaking change. I vote for adding a breaking change and modifying the original methods.